src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -159,60 +158,46 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
-                case SimpleLambdaExpressionSyntax simpleLambda:
-                    return GenerateSimpleLambdaParameterList(semanticModel, simpleLambda, delegateType.DelegateInvokeMethod, cancellationToken);
-                case ParenthesizedLambdaExpressionSyntax parenthesizedLambda:
-                    return GenerateParenthesizedLambdaParameterList(semanticModel, parenthesizedLambda, delegateType.DelegateInvokeMethod, cancellationToken);
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(anonymousFunction);
-            }
-        }
+            var parameterList = GetOrMakeParameterList(anonymousFunction);
+            int i = 0;
 
-        private ParameterListSyntax GenerateSimpleLambdaParameterList(
-            SemanticModel semanticModel, SimpleLambdaExpressionSyntax lambdaExpression, IMethodSymbol delegateInvokeMethod, CancellationToken cancellationToken)
-        {
-            var parameter = semanticModel.GetDeclaredSymbol(lambdaExpression.Parameter, cancellationToken);
-            var type = parameter?.Type.GenerateTypeSyntax() ?? s_objectType;
+            return parameterList != null
+                ? parameterList.ReplaceNodes(parameterList.Parameters, (parameterNode, _) => PromoteParameter(parameterNode, delegateMethod.Parameters[i++]))
+                : SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(delegateMethod.Parameters.Select(parameter =>
+                    PromoteParameter(SyntaxFactory.Parameter(parameter.Name.ToIdentifierToken()), parameter))));
 
-            var parameterSyntax = SyntaxFactory.Parameter(lambdaExpression.Parameter.Identifier).WithType(type);
-            var param = delegateInvokeMethod.Parameters[0];
-            if (param.HasExplicitDefaultValue)
+            ParameterSyntax PromoteParameter(ParameterSyntax parameterNode, IParameterSymbol delegateParameter)
             {
-                parameterSyntax = parameterSyntax.WithDefault(GetDefaultValue(param));
-            }
+                if (parameterNode.Type == null)
+                {
+                    parameterNode = parameterNode.WithType(delegateParameter.Type.GenerateTypeSyntax() ?? s_objectType);
+                }
 
-            return SyntaxFactory.ParameterList(
-                SyntaxFactory.SeparatedList<ParameterSyntax>().Add(parameterSyntax));
+                if (delegateParameter.HasExplicitDefaultValue)
+                {
+                    parameterNode = parameterNode.WithDefault(GetDefaultValue(delegateParameter));
+                }
+
+                return parameterNode;
+            }
         }
 
-        private ParameterListSyntax GenerateParenthesizedLambdaParameterList(
-            SemanticModel semanticModel, ParenthesizedLambdaExpressionSyntax lambdaExpression, IMethodSymbol delegateInvokeMethod, CancellationToken cancellationToken)
+        private static ParameterListSyntax GetOrMakeParameterList(AnonymousFunctionExpressionSyntax anonymousFunction)
```

Comment:
not a great name... hard to come up with one, because this thing actually does return null if it's an anonymous method with no parameter list - i concluded that this case is really fundamentally different because it isn't equivalent to having an empty parameter list there, this should really be a special case

Reply:
So, originally, we had separate helpers to deal with the simple lambda and parenthesized lambda cases.  I don't see why it would be that bad to have 3 helpers here, maybe with sub helpers that are shared.

Reply:
Also, in this case, why not always return a parameter list, even for anonymous-delegates.  if they don't ahve an existing parameter list, then just synthesize one.  That's what we do for simple-lambdas.

Reply:
@CyrusNajmabadi that would be the wrong thing to do because as I sad a missing parameter list is not equivalent to an empty one there. The delegate method may have a lot of parameters after all, so I have to check if this returns null and in that case forget about this and synthesize parameters from the delegate method.

With simple lambdas this is OK because there's no difference between a simple lambda and a parenthesized one with the same argument.

Reply:
Ah, i see what you mean.  Yes, an empty parameter list is wrong.  You shoudl synthesize one based on the types/names of hte delegate Invoke method.

Reply:
it's already implemented, you can see a new test for it

Reply:
I think my point is: do that in this method.  i.e. call the method GetOrCreateParameterList.  In the case of an anonymous-delegate with no parameter list, just do the 'Create' part here (possibly using helpers you have elsewhere).

Reply:
I'd prefer not to.
1. this is sort of a general purpose method that could be used to extract a parameter list from lambdas/delegates
2. right now both cases are handled right next to each other on the next line. that's how i like it, they won't deviate as much from each other

Reply:
if what you mean is just empty an empty parameter list and then check if it's empty... again, that's wrong because an empty one isn't the same as a missing one.

Reply:
in broken code like this:
```c#
Action<string> a = delegate() {};
```
I convert it to a local function taking no parameters (the code is already broken anyway). But if you omit the `()`, I do create a parameter list for you.

Reply:
> if what you mean is just return an empty parameter list and then check if it's empty and fill it with something... 

Definitely not what i meant :)

> I'd prefer not to.

Your call :)  It was just a suggestion.